snapcraft/plugins/gulp.py

```diff
@@ -0,0 +1,107 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2016 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""The gulp plugin drives gulp.js, the streaming build system.
+
+The plugin uses gulp to drive the build. It requires a gulpfile.js in
+the root of the source.
+
+This plugin uses the common plugin keywords as well as those for "sources".
+For more information check the 'plugins' topic for the former and the
+'sources' topic for the latter.
+
+Additionally, this plugin uses the following plugin-specific keywords:
+
+    - gulp-tasks:
+      (list)
+      A list of gulp tasks to run.
+    - node-engine:
+      (string)
+      The version of nodejs you want the snap to run on.
+"""
+
+import logging
+import os
+import shutil
+
+import snapcraft
+from snapcraft import sources
+from snapcraft.plugins import nodejs
+
+logger = logging.getLogger(__name__)
+
+
+class GulpPlugin(snapcraft.BasePlugin):
+
+    @classmethod
+    def schema(cls):
+        schema = super().schema()
+        node_properties = nodejs.NodePlugin.schema()['properties']
+
+        schema['properties']['gulp-tasks'] = {
+            'type': 'array',
+            'minitems': 1,
+            'uniqueItems': True,
+            'items': {
+                'type': 'string'
+            },
+            'default': [],
+        }
+        schema['properties']['node-engine'] = node_properties['node-engine']
+        schema['required'] = ['gulp-tasks']
+
+        # Inform Snapcraft of the properties associated with building. If these
+        # change in the YAML Snapcraft will consider the build step dirty.
+        schema['build-properties'] = ['gulp-tasks']
```

Comment:
Ditto the `source` comment, which makes me think that is indeed what you intended.